e2e/LongHaul/module/IotHub.cs

```diff
@@ -259,6 +261,58 @@ public async Task SetPropertiesAsync(string keyName, object properties, Logger l
             }
         }
 
+        public async Task InvokeDirectMethodOnLeafClientThroughEdgeAsync(string deviceId, string moduleId, string methodName, Logger logger, CancellationToken ct)
+        {
+            Debug.Assert(_moduleClient != null);
+            Debug.Assert(deviceId != null);
+            Debug.Assert(moduleId != null);
+
+            var sw = new Stopwatch();
+            while (!ct.IsCancellationRequested)
+            {
+                try
+                {
+                    var payload = new CustomDirectMethodPayload
+                    {
+                        RandomId = Guid.NewGuid(),
+                        CurrentTimeUtc = DateTimeOffset.UtcNow,
+                        MethodCallsCount = ++_totalMethodCallsModuleToModuleSentCount,
```

Comment:
The increment here may be premature. If we fail due to some exception and try again, we aren't trying with the same payload and increment yet again.